plugins/woocommerce-blocks/assets/js/data/cart/actions.ts

```diff
@@ -188,16 +188,36 @@ export const shippingRatesBeingSelected = ( isResolving: boolean ) =>
  */
 export const applyExtensionCartUpdate =
 	( args: ExtensionCartUpdateArgs ) =>
-	async ( { dispatch }: { dispatch: CartDispatchFromMap } ) => {
+	async ( {
+		dispatch,
+		select,
+	}: {
+		select: CartSelectFromMap;
+		dispatch: CartDispatchFromMap;
+	} ) => {
 		try {
 			const { response } = await apiFetchWithHeaders( {
 				path: '/wc/store/v1/cart/extensions',
 				method: 'POST',
 				data: { namespace: args.namespace, data: args.data },
 				cache: 'no-store',
 			} );
-			dispatch.receiveCart( response );
-			return response;
+			if ( args.overwriteDirtyCustomerData === true ) {
```

Comment:
If you overwrite data here, what happens if there was a dirty state before? Will it push those dirty updates *next* update, or use these new values?

Reply:
The old "dirty" values will be discarded and the new values will be used. I don't think anything will be pushed until the addresses change again.